node/src/service/nonce_monitor.rs

```diff
@@ -135,15 +179,27 @@ pub(super) async fn task(
 		"the nonce for the authority in onchain storage",
 	);
 
-	let acc = AccountId::from_string(&nonce_account)
-		.expect("Invalid account ID provided for nonce monitoring");
-	let key = get_off_chain_nonce_key(&handlers, &acc)
+	let nonce_account =
+		loop {
+			match get_authority_account(monitor_target.clone(), &keystore).await {
+				Ok(Some(acct)) => break acct,
+				Ok(None) => tokio::time::sleep(POLL_INTERVAL * 2).await,
+				Err(e) => {
+					log::error!("Encountered error when trying to get authority account for monitoring: {e}");
+					return;
+				},
+			}
+		};
+
+	let key = get_off_chain_nonce_key(&handlers, &nonce_account)
 		.await
 		.expect("Failed to get key for the offchain nonce");
```

Comment:
Hmm I think it can only fail if the RPC server is down or the runtime is unavailable, but since I'm not 100% sure I replaced it with an error log + return (which would stop monitoring but not panic).